### DISCUSSION TITLE
Possible website rebuild

### DISCUSSION THREAD

**Discussion by theDanielJLewis** [theDanielJLewis/podcasting2.org#10] [Ideas]
Right now, Podcasting2.org is built with Next.js, which is a particular structure for React code. But I just learned about [Starlight](https://starlight.astro.build/guides/sidebar/), which is a different structure for React code, created specifically for documentation websites.

What I like about it is that making and editing documentation could be as simple as almost pure markdown.

While this might mean ditching the current JSON structure for Podcast Namespace tag documentation, I might be able to build something that converts from markdown into JSON structured data through a RESTful API in case anyone needs it.

I'm only thinking about this now. I know Next.js better and I can do more with it, but making something easier for the community to edit would seem more important.

**Comment by rpodcast** [comment]
Just chiming in that I really like this suggestion. For my tool/package documentation sites, I've had great success with markdown as the source format. In particular, I've used both the [Hugo](https://gohugo.io/) generator (with the [relearn](https://themes.gohugo.io/themes/hugo-theme-relearn/) theme) as well as the [Quarto](https://quarto.org/) publishing system with great success. Having a little link on each page that takes the viewer directly to the source file on GitHub would be a great friction-less way to get additional contributions.

**Reply by theDanielJLewis** [reply]
Ooh! I love the suggestion of linking directly to the source for editing!

**Comment by rpodcast** [comment]
> Ooh! I love the suggestion of linking directly to the source for editing!

It's been an extremely useful feature for my sites at the day job. Also here is a workshop site I built last year using the aforementioned Quarto and the edit links are in the right sidebar: https://posit-conf-2023.github.io/shiny-r-prod/
